src/test/java/com/minecolonies/util/DistanceSquaredTest.java

```diff
@@ -0,0 +1,71 @@
+package com.minecolonies.util;
+
+import com.minecolonies.colony.ColonyManager;
+import net.minecraft.util.BlockPos;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.number.OrderingComparison.greaterThanOrEqualTo;
+
+@PrepareForTest({ColonyManager.class, LanguageHandler.class})
+@RunWith(PowerMockRunner.class)
+public class DistanceSquaredTest
+{
+
+    /**
+     * Max x/z position in vanilla minecraft.
+     */
+    private static final int MAX_POSITION =  30_000_000;
+
+    @Test
+    public void testDistance2DTwoPositions()
+    {
+        for (int i = -400; i < 400; i+=20)
+        {
+            BlockPos posA = new BlockPos(i^2, 0, i*2);
```

Comment:
5 distances. Honestly if you tested the max minecraft distance and verified that the result was equal to what you expect, that should be fine.